heartbeat/azure-phoenix.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/python
+
+import sys, re, os, subprocess, time, syslog
+import logging
+import azure_fence_lib
+
+OCF_SUCCESS = 0
+OCF_ERR_GENERIC = 1
+OCF_ERR_UNIMPLEMENTED = 3
+OCF_ERR_CONFIGURED = 6
+OCF_NOT_RUNNING = 7
+
+AZR_VM_FENCED = 1
+AZR_VM_NOT_FENCED = 2
+
+OCF_RESOURCE_INSTANCE = None
+PROCESS_EXEC_NAME = None
+PROCESS_EXEC_ARG = None
+PID_FILE = "azure-phoenix-{}.pid"
+
+## https://github.com/ClusterLabs/fence-agents/blob/master/fence/agents/lib/fencing.py.py
+## Own logger handler that uses old-style syslog handler as otherwise everything is sourced
+## from /dev/syslog
+class SyslogLibHandler(logging.StreamHandler):
+	"""
+	A handler class that correctly push messages into syslog
+	"""
+	def emit(self, record):
+		syslog_level = {
+			logging.CRITICAL:syslog.LOG_CRIT,
+			logging.ERROR:syslog.LOG_ERR,
+			logging.WARNING:syslog.LOG_WARNING,
+			logging.INFO:syslog.LOG_INFO,
+			logging.DEBUG:syslog.LOG_DEBUG,
+			logging.NOTSET:syslog.LOG_DEBUG,
+		}[record.levelno]
+
+		msg = self.format(record)
+
+		# syslos.syslog can not have 0x00 character inside or exception is thrown
+		syslog.syslog(syslog_level, msg.replace("\x00", "\n"))
+		return
+
+def get_azure_config():
+    config = azure_fence_lib.AzureConfiguration()
+
+    config.RGName = os.environ.get("OCF_RESKEY_resourceGroup")
+    config.VMName = os.environ.get("OCF_RESKEY_vmName")
+    config.SubscriptionId = os.environ.get("OCF_RESKEY_subscriptionId")
+    config.Cloud = os.environ.get("OCF_RESKEY_cloud")
+    config.UseMSI = os.environ.get("OCF_RESKEY_useMSI")
+    config.Tenantid = os.environ.get("OCF_RESKEY_tenantId")
+    config.ApplicationId = os.environ.get("OCF_RESKEY_applicationId")
+    config.ApplicationKey = os.environ.get("OCF_RESKEY_applicationKey")
+    config.Verbose = os.environ.get("OCF_RESKEY_verbose")
+
+    if len(sys.argv) > 2:
```

Comment:
You can remove this entire section, as the above os.environ.get()'s will get them when you do e.g. 'OCF_RESKEY_resourceGroup="rg" azure-phoenix.py', which is how the resource agents are supposed to be run.